python/text/textindex.py

```diff
@@ -229,11 +250,27 @@ def extract_descriptions(input_filename, index, texts):
 
 
 # [START get_text]
-def get_text_from_file(vision, index, input_filename):
+def get_text_from_files(vision, index, input_filenames):
     """Call the Vision API on a file and index the results."""
-    with open(input_filename, 'rb') as image:
-        texts = vision.detect_text(image)
-        extract_descriptions(input_filename, index, texts)
+    texts = vision.detect_text(input_filenames)
+    for filename, text in texts.items():
+        extract_descriptions(filename, index, text)
+
+
+def batch(iterable, batch_size=4):
```

Comment:
Sounds good, I can make it a constant and bump it up.

Probably the best thing would be to batch based on size (get as close to 8MB limit as possible). I can see if that would be hard to do. I imagine string length of the base64 string could be useful for this.